### ISSUE TITLE
Hidden perk preset in dropdown menu can be prematurely revealed via keyboard navigation.

### ISSUE THREAD

**Issue by TacoRat2** [Grimy/Grimy.github.io#99]
Been using this tool for a while. Just unlocked the Trimp challenge and I decided to peep the wiki for strategy tips and I found mention of a Trimp² preset on the [wiki](https://trimps.fandom.com/wiki/Trimp_(challenge)). 

Didn't find a Trimp² preset under the preset dropdown menu, but used keyboard navigation, pressing 'T', revealing spoilers. 

Peeped around source and discovered certain presets are meant to remain hidden until the save file has progressed far enough. This doesn't account for keyboard navigation, thus these hidden presets can be accessible and selectable before it should be visible. 

Steps to reproduce (Video attached)
https://github.com/user-attachments/assets/0984aff9-11dc-43c1-aeec-3ad9d2e93b14

1. Input a save string that has not yet unlocked the presets tagged with the 'toggle_spoilers' function.
2. Open Presets dropdown menu.
3.  With the presets dropdown focused, press a key that corresponds to a hidden present (e.g., 'T' for the Trimp² challenge)
4. Observe.

Is it possible to remove hidden presets entirely until unlocked, instead of hiding them with CSS?

P.S
Thank you for making this tool, it's been super helpful for helium distribution and learning the game. HAGD
